lib/engine/game/g_18_usa/game.rb

```diff
@@ -1605,32 +1628,34 @@ def setup
           city_by_id('CTownX-0-0').place_token(neutral, neutral.next_token)
           city_by_id('CTownY-0-0').place_token(neutral, neutral.next_token)
 
-          metro = METROPOLITAN_HEXES.sort_by { rand }.take(3)
-          metro.each do |i|
-            hex = @hexes.find { |h| h.id == i }
-            active_metropolitan_hexes << hex
-            case i
-            when 'H14'
-              hex.lay(@tiles.find { |t| t.name == 'DFW1' })
-            when 'E11'
-              # Denver needs to be done at a later date
-              hex.lay(@tiles.find { |t| t.name == 'D0' })
-              @metro_denver = true
-            when 'G3'
-              hex.lay(@tiles.find { |t| t.name == 'LA1' }.rotate!(3))
-            when 'D20'
-              hex.lay(@tiles.find { |t| t.name == 'CHI1' }.rotate!(1))
-            when 'I19'
-              hex.lay(@tiles.find { |t| t.name == 'NO1' })
-              @metro_new_orleans = true
-            when 'H22'
-              hex.lay(@tiles.find { |t| t.name == 'ATL1' })
-            end
-          end
+          metro_hexes = METROPOLITAN_HEXES.sort_by { rand }.take(3)
+          metro_hexes.each { |metro_hex| convert_potential_metro(metro_hex) }
 
           randomize_subsidies
         end
 
+        # Convert a potential metro hex to a metro hex
+        def convert_potential_metro(metro_hex_id)
```

Comment:
Based on the other comments, I think you can make this method take in the hex object and then not have to look it up.

Reply:
Changed